### PR TITLE
Create opinionated EKS Core module

### PR DESCRIPTION
**Pull request by jimgus**
In order to be able to add network ACLs to the core setup of EKS, an EKS opinionated core module was created and the previous "generic" core module was renamed to "generic-core". Names can of course be discussed here. The new EKS opinionated core module contains copied code from "generic-core" . It would however be possible to refer to the "generic-core" module from this module but the reason for not doing that is that we allow the different code bases to diverge if needed. Could of course also be discussed.

### CODE REVIEW

File: modules/aws/core/vpc.tf [hunk 1]
```diff
@@ -277,3 +439,179 @@ resource "aws_route" "peering_accepter" {
...
+
+  tags = merge(
+    local.global_tags,
+    {
+      Name = each.key,
+    },
+  )
+}
+
+# Allow all from private, public and eks1-cluster subnets
```
Comment (NissesSenap) [inline modules/aws/core/vpc.tf:504]: I'm probably messing things up but don't you use allow local.private_subnets that you just deny?
Or is this to make it possible to talk from the public network to the private?
If so can we clarify the comment a bit.

File: modules/aws/generic-core/main.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+locals {
+  global_tags = {
+    Name        = var.name
+    Environment = var.environment
+    Module      = "core"
```
Comment (NissesSenap) [inline modules/aws/generic-core/main.tf:26]: Are we in core or in generic-core?

**Review by NissesSenap — COMMENTED** [review]
First of all great job!
It's kind of a big change so I will probably need a few times to read it. I have left some small comments where i probably don't understand.

About the name generic-core I have nothing against it.
I think it would be good to sit down offline and decide if we should have duplicate code or if we should depend on generic-core instead.

File: modules/aws/core/README.md [hunk 3]
```diff
@@ -54,15 +60,17 @@ No modules.
 
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
-| <a name="input_dns_zone"></a> [dns\_zone](#input\_dns\_zone) | The DNS Zone host name | `string` | `""` | no |
-| <a name="input_dns_zone_enabled"></a> [dns\_zone\_enabled](#input\_dns\_zone\_enabled) | Should dns zone be enabled | `bool` | `false` | no |
-| <a name="input_environment"></a> [environment](#input\_environment) | The environment name to use for the deploy | `string` | n/a | yes |
+| <a name="input_cidr_block"></a> [cidr\_block](#input\_cidr\_block) | CIDR block of the VPC. The prefix lenght of the CIDR block must be 18 or less | `string` | n/a | yes |
```
Comment (ahockersten) [inline modules/aws/core/README.md:63]:
```suggestion
| <a name="input_cidr_block"></a> [cidr\_block](#input\_cidr\_block) | CIDR block of the VPC. The prefix length of the CIDR block must be 18 or less | `string` | n/a | yes |
```
spelling

File: modules/aws/core/README.md [hunk 4]
```diff
@@ -54,15 +60,17 @@ No modules.
...
+| <a name="input_name"></a> [name](#input\_name) | Deployment name | `string` | n/a | yes |
 | <a name="input_vpc_peering_config_accepter"></a> [vpc\_peering\_config\_accepter](#input\_vpc\_peering\_config\_accepter) | VPC Peering configuration accepter | <pre>list(object({<br>    name                   = string<br>    peer_owner_id          = string<br>    destination_cidr_block = string<br>  }))</pre> | `[]` | no |
 | <a name="input_vpc_peering_config_requester"></a> [vpc\_peering\_config\_requester](#input\_vpc\_peering\_config\_requester) | VPC Peering configuration requester | <pre>list(object({<br>    name                   = string<br>    peer_owner_id          = string<br>    peer_vpc_id            = string<br>    destination_cidr_block = string<br>  }))</pre> | `[]` | no |
 
 ## Outputs
 
-No outputs.
+| Name | Description |
+|------|-------------|
+| <a name="output_private_subnets_ids"></a> [private\_subnets\_ids](#output\_private\_subnets\_ids) | The ids of the of private subnets created by this module |
```
Comment (ahockersten) [inline modules/aws/core/README.md:75]:
```suggestion
| <a name="output_private_subnets_ids"></a> [private\_subnets\_ids](#output\_private\_subnets\_ids) | The ids of the private subnets created by this module |
```

File: modules/aws/core/variables.tf [hunk 5]
```diff
@@ -1,23 +1,21 @@
...
   type        = string
-  default     = ""
 }
 
-variable "dns_zone_enabled" {
-  description = "Should dns zone be enabled"
-  type        = bool
-  default     = false
+variable "cidr_block" {
+  description = "CIDR block of the VPC. The prefix lenght of the CIDR block must be 18 or less"
```
Comment (ahockersten) [inline modules/aws/core/variables.tf:17]:
```suggestion
  description = "CIDR block of the VPC. The prefix length of the CIDR block must be 18 or less"
```
spelling

File: modules/aws/core/vpc.tf [hunk 6]
```diff
@@ -277,3 +439,179 @@ resource "aws_route" "peering_accepter" {
   destination_cidr_block    = each.value.destination_cidr_block
   vpc_peering_connection_id = each.value.vpc_peering_connection_id
 }
+
+
+
+# Allow all from eks1-nodes and eks2-nodes subnets to enable outbound
+# internet access from PODs trough NAT Gateway
```
Comment (ahockersten) [inline modules/aws/core/vpc.tf:446]: Nit
```suggestion
# internet access from PODs through NAT Gateway
```

File: modules/aws/generic-core/vpc.tf [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+  for_each = {
+    for idx, subnet in var.vpc_config.public_subnets :
+    "${local.public_subnet_name_prefix}-${idx}" => subnet
+  }
+
+  subnet_id      = aws_subnet.public[each.key].id
+  route_table_id = aws_route_table.public[each.key].id
+}
+
+# Private subnetes with egress routing through NAT
```
Comment (ahockersten) [inline modules/aws/generic-core/vpc.tf:142]:
```suggestion
# Private subnets with egress routing through NAT
```

File: modules/aws/core/vpc.tf (hunk 1 above)
Comment (jimgus) [reply modules/aws/core/vpc.tf:504]: I think that the fact that we have a subnet named "private" and a number of subnets of type "private" (eks*_nodes, eks*_cluster, private) is messing things up a bit. At least it did for me :-) This ACL is allowing traffic from the subnets named "private", "public" and "eks1_cluster" to "eks1_nodes". I think I should add more documentation to this module visualizing the network setup and the allowed traffic flow between the subnets

**Comment by jimgus** [comment]
At offline review we decided to call the new module "eks-core" and that it shall not have a dependency to the generic "core" module. This PR will not contain any subnet ACLs since we wan't to take it in steps so an additional PR will come. Also, more documentation will also be added in a separate PR.

File: modules/aws/core/vpc.tf (hunk 6 above)
Comment (jimgus) [reply modules/aws/core/vpc.tf]: Fixed

**Review by phillebaba — APPROVED** [review]
LGTM :rose: 

Good first step towards making this module more opinionated

**Pull request merged**